services/wireguard/consumer.go

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package wireguard
+
+import (
+	"errors"
+	"net"
+
+	"github.com/mdlayher/wireguardctrl/wgtypes"
+)
+
+var errNoIPAddress = errors.New("IP address required")
+
+type consumer struct {
+	ip         net.IPNet
+	peer       wgtypes.PeerConfig
+	privateKey wgtypes.Key
+}
+
+func (c consumer) Config() (serviceConfig, error) {
+	if len(c.peer.AllowedIPs) == 0 {
+		return serviceConfig{}, errNoIPAddress
+	}
+
+	var config serviceConfig
+
+	config.Provider.IP = c.peer.AllowedIPs[0]
```

Comment:
Not sure, but I think its incorrect. Its provider configuration as seen by Consumer? 
Why its called Provider.IP ? It should be AllowedIPs (multi) meaning which IPs will be routed towards WG peer (provider).

Reply:
Also `c.peer.AllowedIPs[0]` probably would give here incorrect value. Here it almost always should be 0.0.0.0/0 in our case.

Reply:
We could have more extensive explanation here. To remove ambiguity how these fields are / should be used.

Reply:
We are generating `serviceConfig` here, that will be sent to the consumer to establish a connection. 
This config has two blocks: Provider part and Consumer part.

`c.peer` is a consumer's peer - Provider. That is why it goes to `config.Provider` to be sent as service configuration.

Reply:
`c.peer.AllowedIPs[0]` is a Provider local IP address, it's not Consumer IP. It's not empty and needed to allow customer know how to connect to the Provider after tunnel established.

Getting only one addess since do not think that we need to support multiple IP addresses at that moment. One is enough for now.

Reply:
For consumer to reach provider, provider's endpoint should be enough. This Provider.IP should not be local provider IP.

Reply:
Lets keep terms the same as they are used by WG.
We have 3 parameters describing Provider at Consumer side: 
`[Peer]
Endpoint = 94.237.xx.xx:51820
PublicKey = changed here.. 
AllowedIPs = 0.0.0.0/0, ::/0
`

Reply:
please see here: 
https://wiki.archlinux.org/index.php/WireGuard 
(Specific use-case: setup a VPN server)

For better described example.

Reply:
Removed `IP` and now sending only subnets called AllowedIPs.